### DISCUSSION TITLE
Cannot Sign Up or Sign In

### DISCUSSION THREAD

**Discussion by IzMichael** [pocketbase/pocketbase#344] [Q&A] [answered]
Hiya, whenever I try to sign up/in with the REST api, it returns a 400 error with the following message:
```JSON
{
    "code": 400,
    "message": "Failed to load the submitted data due to invalid formatting.",
    "data": {}
}
```
I'm not sure where the formatting error is, but here is my login code (email and password redacted):
```JS
let user = await fetch('https://db.**********.co.nz/api/users/auth-via-email', {
    "method": "POST",
    "body": JSON.stringify({
        "email": "*****@*****.com",
        "password": "************"
    })
});
```
My signup code is much the same:
```JS
let user = await fetch('https://db.**********.co.nz/api/users', {
    "method": "POST",
    "body": JSON.stringify({
        "email": "*****@*****.com",
        "password": "************",
        "passwordConfirm": "************",
    })
});
```

Thanks for any help!

**Comment by ganigeorgiev** [comment, marked as answer]
You need to provide a `'Content-Type': 'application/json'` header to your `fetch` options when sending JSON:
```js
let authData = await fetch('https://db.therosyshell.co.nz/api/users/auth-via-email', {
    "method": "POST",
    "headers": {
        'Content-Type': 'application/json'
    },
    "body": JSON.stringify({
        "email": "*****@*****.com",
        "password": "************"
    })
});
```

This is all seamlessly handled by the [JavaScript-SDK](https://github.com/pocketbase/js-sdk):

```js
import PocketBase from 'pocketbase';

const client = new PocketBase('https://db.therosyshell.co.nz');

const authData = await client.users.authViaEmail('*****@*****.com', '************');
```

**Reply by IzMichael** [reply]
oh wow, stupid mistake lol. ty!

**Reply by agingaRichard** [reply]
You can do that and still get a 400 error... In that case, the problem could be that you have not created a 'passwordConfirm' field, which is required before you can create a user.

**Reply by ganigeorgiev** [reply]
@agingaRichard You can access the validation errors by `try/catch`-ing your call and inspecting the error response, eg. `console.log(err.data)` (if you are using the JS SDK you can check https://github.com/pocketbase/js-sdk#errors-handling)
